### PR TITLE
Gmsh tags - 2d

### PR DESCRIPTION
**Pull request by alessiofumagalli**
- the boundary in gmsh are physically tagged
- the boundary, fractures, and auxiliary physical tags are propagated in pp face tags
everything works only in 2d. One test is added. In principle all the physical tags associated to faces from gmsh are kept in pp, so far only the standard one are tested.

### CODE REVIEW

File: src/porepy/fracs/simplex.py [hunk 1]
```diff
@@ -277,7 +277,7 @@ def _run_gmsh(file_name, network, **kwargs):
     return pts, cells, cell_info, phys_names
 
 
-def triangle_grid(fracs, domain, do_snap_to_grid=False, **kwargs):
+def triangle_grid(fracs, domain, subdomains, do_snap_to_grid=False, **kwargs):
```
[the remark this answers is not in the document]
Comment (keileg) [reply src/porepy/fracs/simplex.py]: .. I know this is partly old sins, but we need to progress generally.

File: src/porepy/fracs/simplex.py [hunk 2]
```diff
@@ -389,6 +393,8 @@ def triangle_grid(fracs, domain, do_snap_to_grid=False, **kwargs):
     # We find the end points that are shared by more than one intersection
     intersections = __find_intersection_points(lines_split)
 
+    intersections = np.empty(0)
```
Comment (keileg) [inline src/porepy/fracs/simplex.py]: You overwrite the old line? Why? If correct, it is better to replace the first definition of intersections.

File: src/porepy/fracs/simplex.py [hunk 3]
```diff
@@ -618,13 +625,19 @@ def __merge_domain_fracs_2d(dom, frac_p, frac_l):
...
+
     # Merge the point arrays, compartment points first
-    p = np.hstack((dom_p, frac_p))
+    p = np.hstack((dom_p, frac_p, subdom_p))
 
     # Adjust index of fracture points to account for the compartment points
     frac_l[:2] += dom_p.shape[1]
 
-    l = np.hstack((dom_l, frac_l)).astype("int")
+    # Adjust index of subdomains points to account for the compartment points
```
Comment (keileg) [inline src/porepy/fracs/simplex.py]: Terminology: Compartment is old wording, we should probably use subdomains instead.

File: src/porepy/fracs/simplex.py [hunk 4]
```diff
@@ -634,6 +647,23 @@ def __merge_domain_fracs_2d(dom, frac_p, frac_l):
 
 def __find_intersection_points(lines):
```
Comment (keileg) [inline src/porepy/fracs/simplex.py]: While we are at it, I think we should change this and the previous function name to single underscore.

File: src/porepy/grids/gmsh/gmsh_interface.py [hunk 5]
```diff
@@ -92,79 +92,113 @@ def __write_fractures_compartments_2d(self):
...
+                if tag[mask] != constants.AUXILIARY_TAG:
+                    name = "frac_line_"
+                    physical_name = constants.PHYSICAL_NAME_FRACTURES
+                else:
+                    name = "seg_line_"
+                    physical_name = constants.PHYSICAL_NAME_AUXILIARY
+
                 s += (
-                    "frac_line_"
+                    name
```
Comment (keileg) [inline src/porepy/grids/gmsh/gmsh_interface.py]: I am not sure I agree with black's splitting of this one, but I suppose we're not entitled to such opinions.

File: src/porepy/grids/gmsh/mesh_2_grid.py [hunk 6]
```diff
@@ -28,9 +28,9 @@ def create_2d_grids(pts, cells, **kwargs):
     # List of 2D grids, one for each surface
     g_2d = []
     is_embedded = kwargs.get("is_embedded", False)
+    phys_names = kwargs.get("phys_names", False)
+    cell_info = kwargs.get("cell_info", False)
     if is_embedded:
-        phys_names = kwargs.get("phys_names", False)
```
Comment (keileg) [inline src/porepy/grids/gmsh/mesh_2_grid.py]: Just an observation: For embedded 2d grids (dfn), the new boundary tag treatment is not fully developed?

File: test/unit/test_tag_gmsh.py [hunk 7]
```diff
@@ -0,0 +1,325 @@
+import numpy as np
```
Comment (keileg) [inline test/unit/test_tag_gmsh.py]: Please add short documentation of what the module is testing. This we should really for all tests.

File: src/porepy/grids/gmsh/mesh_2_grid.py [hunk 8]
```diff
@@ -113,10 +113,34 @@ def create_2d_grids(pts, cells, **kwargs):
             g_2d.append(g)
 
     else:
+
         triangles = cells["triangle"].transpose()
         # Construct grid
         g_2d = simplex.TriangleGrid(pts.transpose(), triangles)
 
+        # save all the extra tags for the grids
```
Comment (keileg) [inline src/porepy/grids/gmsh/mesh_2_grid.py]: This is a place where I miss comments on two levels. First, what is this new piece of code doing, on a high level. Second, what is the role of the second for loop.

File: src/porepy/fracs/simplex.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (alessiofumagalli) [reply src/porepy/fracs/simplex.py]: documentation is coming in the next commit which fix also the test

File: src/porepy/fracs/simplex.py (hunk 2 above)
Comment (alessiofumagalli) [reply src/porepy/fracs/simplex.py]: this is definitly wrong, I cherry pick this commit and after there is another one to fix it. I cherry pick it as well

File: src/porepy/fracs/simplex.py (hunk 3 above)
Comment (alessiofumagalli) [reply src/porepy/fracs/simplex.py]: oki

File: src/porepy/fracs/simplex.py (hunk 4 above)
Comment (alessiofumagalli) [reply src/porepy/fracs/simplex.py]: only this function?

File: src/porepy/grids/gmsh/mesh_2_grid.py (hunk 6 above)
Comment (alessiofumagalli) [reply src/porepy/grids/gmsh/mesh_2_grid.py]: I've been doing this part only for 2d not embedded.

File: src/porepy/grids/gmsh/mesh_2_grid.py (hunk 8 above)
Comment (alessiofumagalli) [reply src/porepy/grids/gmsh/mesh_2_grid.py]: OK, done.

**Comment by alessiofumagalli** [comment]
@keileg , I think I finished with your comments.

**Pull request merged**